Description: Allow display overrides via ~/.Xresources
Bug: http://bugs.debian.org/727796
Index: grace-5.1.25/src/xmgrace.c
===================================================================
--- grace-5.1.25.orig/src/xmgrace.c
+++ grace-5.1.25/src/xmgrace.c
@@ -211,7 +211,7 @@ static XtResource resources[] =
 String fallbackResourcesCommon[] = {
     "XMgrace.consoleDialog*text.columns: 72",
     "XMgrace.consoleDialog*text.rows: 5",
-    "XMgrace*background: #e5e5e5",
+    /*"XMgrace*background: #e5e5e5",
     "XMgrace*foreground: #000000",
     "XMgrace*XbaeMatrix.oddRowBackground: #cccccc",
     "XMgrace*XbaeMatrix.evenRowBackground: #cfe7e7",
@@ -223,18 +223,18 @@ String fallbackResourcesCommon[] = {
     "XMgrace*XmPushButton.background: #b0c4de",
     "XMgrace*XmMenuShell*XmPushButton.background: #e5e5e5",
     "XMgrace*XmText*background: #cfe7e7",
-    "XMgrace*XmToggleButton.selectColor: #ff0000",
+    "XMgrace*XmToggleButton.selectColor: #ff0000",*/
     "XMgrace*XmToggleButton.fillOnSelect: true",
     "XMgrace*XmSeparator.margin: 0",
 #ifdef WITH_XMHTML
-    "XMgrace*XmHTML.background: #ffffff",
-    "XMgrace*XmHTML.foreground: #000000",
+/*    "XMgrace*XmHTML.background: #ffffff",
+    "XMgrace*XmHTML.foreground: #000000", */
     "XMgrace*XmHTML.width: 600",
     "XMgrace*XmHTML.height: 500",
 #endif
-    "XMgrace*mainWin.shadowThickness: 0",
+/*    "XMgrace*mainWin.shadowThickness: 0",
     "XMgrace*mainWin.menuBar.shadowThickness: 1",
-    "XMgrace*menuBar*tearOffModel: XmTEAR_OFF_ENABLED",
+    "XMgrace*menuBar*tearOffModel: XmTEAR_OFF_ENABLED", */
     "XMgrace*dragInitiatorProtocolStyle: XmDRAG_NONE",
     "XMgrace*dragReceiverProtocolStyle:  XmDRAG_NONE",
     "XMgrace*fileMenu.newButton.acceleratorText: Ctrl+N",
