diff -Nru grace-5.1.25.orig/T1lib/t1lib/t1base.c grace-5.1.25/T1lib/t1lib/t1base.c
--- grace-5.1.25.orig/T1lib/t1lib/t1base.c	2021-12-27 17:31:44.000000000 -0500
+++ grace-5.1.25/T1lib/t1lib/t1base.c	2021-12-27 19:07:02.000000000 -0500
@@ -653,7 +653,7 @@
  
 
 
-/* test_for_t1_file returns 0 if a file "name.pfa" or "name.pfb"
+/* test_for_t1_file returns 0 if a file "name.pfa" or "name.pfb" or "name.t1"
    was found. Else, -1 is returned. If successful, buffer contains the
    found filename string */
 int test_for_t1_file( char *buffer )
@@ -671,6 +671,17 @@
   while (buffer[i]!=0){
     i++;
   }
+  /* One-and-a-halfth case: t1, alternate extension for pfb */
+  /* Check this first in case the buffer is one character shorter from the two-character extension */
+  buffer[i] = '.';
+  buffer[i+1]='t';
+  buffer[i+2]='1';
+  buffer[i+3]='\0';
+  if ((FullName=intT1_Env_GetCompletePath(buffer,T1_PFAB_ptr))!=NULL) {
+    free(FullName);
+    return(0);
+  }
+
   buffer[i]='.';
   buffer[i+1]='p';
   buffer[i+2]='f';
diff -Nru grace-5.1.25.orig/ac-tools/configure.in grace-5.1.25/ac-tools/configure.in
--- grace-5.1.25.orig/ac-tools/configure.in	2021-12-27 17:31:43.000000000 -0500
+++ grace-5.1.25/ac-tools/configure.in	2021-12-27 19:07:15.000000000 -0500
@@ -283,7 +283,7 @@
     dnl **** default settings for gcc
     DEBUG="-g -O2"
     OPTIMIZE="-O2"
-    CFLAGS="-fno-common -Wall -Wpointer-arith -Wnested-externs"
+    CFLAGS="-fno-common -Wall -Wpointer-arith -Wnested-externs -mtune=native -mfpmath=sse -msse -msse2 -msse3 -mmmx -m3dnow"
 
     dnl **** check for strength-reduce bug
     ACX_GCC_STRENGTH_REDUCE(CFLAGS="$CFLAGS -fno-strength-reduce")
diff -Nru grace-5.1.25.orig/src/svgdrv.c grace-5.1.25/src/svgdrv.c
--- grace-5.1.25.orig/src/svgdrv.c	2021-12-27 17:31:44.000000000 -0500
+++ grace-5.1.25/src/svgdrv.c	2021-12-27 19:07:08.000000000 -0500
@@ -745,7 +745,7 @@
             -tm->cxy, -tm->cyy,
             scaleval(vp.x), scaleval(vp.y));
 
-    fprintf(prstream, escape_specials((unsigned char *) s, len));
+    fprintf(prstream, "%s", escape_specials((unsigned char *) s, len));
 
     fprintf(prstream, "</text>\n");
 }
diff -Nru grace-5.1.25.orig/src/utils.c grace-5.1.25/src/utils.c
--- grace-5.1.25.orig/src/utils.c	2021-12-27 17:31:44.000000000 -0500
+++ grace-5.1.25/src/utils.c	2021-12-27 19:07:08.000000000 -0500
@@ -1171,12 +1171,12 @@
     if (inwin) {
         stufftextwin(s);
     } else {
-        printf(s);
+        printf("%s", s);
     }
 #endif
     /* log results to file */
     if (resfp != NULL) {
-	fprintf(resfp, s);
+	fprintf(resfp, "%s", s);
     }
 }
 
@@ -1222,7 +1222,9 @@
     char buf[GR_MAXPATHLEN];
     
     if (wd == NULL) {
-        getcwd(workingdir, GR_MAXPATHLEN - 1);
+        if (getcwd(workingdir, GR_MAXPATHLEN - 1) == NULL) {
+	    return RETURN_FAILURE;
+        }
         if (workingdir[strlen(workingdir)-1] != '/') {
             strcat(workingdir, "/");
         }
diff -Nru grace-5.1.25.orig/src/xmgrace.c grace-5.1.25/src/xmgrace.c
--- grace-5.1.25.orig/src/xmgrace.c	2021-12-27 17:31:44.000000000 -0500
+++ grace-5.1.25/src/xmgrace.c	2021-12-27 19:06:56.000000000 -0500
@@ -211,7 +211,7 @@
 String fallbackResourcesCommon[] = {
     "XMgrace.consoleDialog*text.columns: 72",
     "XMgrace.consoleDialog*text.rows: 5",
-    "XMgrace*background: #e5e5e5",
+    /*"XMgrace*background: #e5e5e5",
     "XMgrace*foreground: #000000",
     "XMgrace*XbaeMatrix.oddRowBackground: #cccccc",
     "XMgrace*XbaeMatrix.evenRowBackground: #cfe7e7",
@@ -223,18 +223,18 @@
     "XMgrace*XmPushButton.background: #b0c4de",
     "XMgrace*XmMenuShell*XmPushButton.background: #e5e5e5",
     "XMgrace*XmText*background: #cfe7e7",
-    "XMgrace*XmToggleButton.selectColor: #ff0000",
+    "XMgrace*XmToggleButton.selectColor: #ff0000",*/
     "XMgrace*XmToggleButton.fillOnSelect: true",
     "XMgrace*XmSeparator.margin: 0",
 #ifdef WITH_XMHTML
-    "XMgrace*XmHTML.background: #ffffff",
-    "XMgrace*XmHTML.foreground: #000000",
+/*    "XMgrace*XmHTML.background: #ffffff",
+    "XMgrace*XmHTML.foreground: #000000", */
     "XMgrace*XmHTML.width: 600",
     "XMgrace*XmHTML.height: 500",
 #endif
-    "XMgrace*mainWin.shadowThickness: 0",
+/*    "XMgrace*mainWin.shadowThickness: 0",
     "XMgrace*mainWin.menuBar.shadowThickness: 1",
-    "XMgrace*menuBar*tearOffModel: XmTEAR_OFF_ENABLED",
+    "XMgrace*menuBar*tearOffModel: XmTEAR_OFF_ENABLED", */
     "XMgrace*dragInitiatorProtocolStyle: XmDRAG_NONE",
     "XMgrace*dragReceiverProtocolStyle:  XmDRAG_NONE",
     "XMgrace*fileMenu.newButton.acceleratorText: Ctrl+N",
